Calculator/Calculator/ExpressionParser.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  ExpressionParser.swift
+//  Calculator
+//
+//  Created by Hyungmin Lee on 2023/06/02.
+//
+
+enum ExpressionParser<OperandQueue: Queueable, OperatorQueue: Queueable> where OperandQueue.Element == Double, OperatorQueue.Element == Operator {
+    static func parse (from input: String) -> Formula<OperandQueue, OperatorQueue> {
+        let operatorComponents = input.compactMap { Operator(rawValue: $0) }
+        let operandComponents = componentsByOperators(from: input).compactMap { return Double($0) }
+        var operatorQueue = OperatorQueue()
+        var operandQueue = OperandQueue()
+        
+        operatorComponents.forEach{ operatorQueue.enqueue(element: $0) }
+        operandComponents.forEach { operandQueue.enqueue(element: $0) }
+        let formula = Formula(operands: operandQueue, operators: operatorQueue)
+        
+        return formula
```

Comment:
사바사겠지만, 현재 고차함수를 활용하는 부분은 체이닝 단위로 개행하고, 한번에 처리하는 것이 가독성 측면에서 더 보기 좋다고 생각합니다. 굳이 인스턴스로 만든다음 다시 forEach를 돌 필요는 없다고 생각해요.
 다음과 같이 수정하는 건 어떤가요? 😁
```suggestion
        var operatorQueue = OperatorQueue()
        var operandQueue = OperandQueue()
        
        input
            .compactMap { Operator(rawValue: $0) }
            .forEach { operatorQueue.enqueue(element: $0) }
        
        componentsByOperators(from: input)
            .compactMap { Double($0) }
            .forEach { operandQueue.enqueue(element: $0) }
        
        let formula = Formula(operands: operandQueue, operators: operatorQueue)
        
        return formula
```

Reply:
가독성 측면에서 훨씬 깔끔하네요 원비 조언 감사합니다^^